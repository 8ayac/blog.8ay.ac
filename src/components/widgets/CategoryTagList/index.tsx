import { CategoryTag } from '@/src/components/elements/CategoryTag';
import { FlexContainerDiv } from '@/src/shared/styles/abstractStyledComponents';
import styled from '@emotion/styled';
import { faTags } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled(FlexContainerDiv)`
    flex-flow: row wrap;

    > * {
      margin-right: 0.7rem;
      margin-bottom: 0.7rem;
    }
  `,

  PreIconWrapperDiv: styled(FlexContainerDiv)`
    align-items: center;
    color: ${(props) => props.theme.color.text.primaryLight};
  `,
};

export const CategoryTagList: React.FC<{
  tags: string[];
  withPreIcon?: boolean;
}> = ({ tags, withPreIcon = false }) => (
  <>
    <S.ComponentWrapperDiv>
      {withPreIcon && (
        <S.PreIconWrapperDiv>
          <FontAwesomeIcon icon={faTags} />
        </S.PreIconWrapperDiv>
      )}
      {[...new Set(tags)].sort().map((tag) => (
        <CategoryTag key={tag} name={tag} />
      ))}
    </S.ComponentWrapperDiv>
  </>
);
