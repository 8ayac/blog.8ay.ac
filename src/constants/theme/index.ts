import * as colors from '@/src/constants/colors';

export const theme = {
  color: {
    blue: colors.blue,
    green: colors.green,
    yellow: colors.yellow,

    primary: colors.blue.base,
    primaryLight: colors.blue.light,
    primaryDark: colors.blue.dark,

    text: {
      primary: 'hsla(0, 0%, 15%)',
      primaryLight: 'hsla(0, 0%, 40%)',
      secondary: colors.blue.dark2,
      accent: colors.green.dark,
    },
  },
};
