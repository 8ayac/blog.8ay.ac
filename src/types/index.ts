export type ArticleAttribute = {
  title: string;
  publishedAt: Date;
  updatedAt: Date;
  tags?: string[];
};

export type Article = ArticleAttribute & {
  body: string;
};
