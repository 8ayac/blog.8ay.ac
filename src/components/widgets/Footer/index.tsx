import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import React from 'react';

const S = {
  Footer: styled.footer`
    padding: 3rem 0;
    font-size: 1.5rem;
    text-align: center;
  `,

  CopyRightParagraph: styled.p``,

  AboutGAParagraph: styled.p`
    color: ${theme.color.text.primaryLight};
  `,
};

export const Footer: React.FC = () => (
  <S.Footer>
    <S.CopyRightParagraph>
      Â© 2021 blog.8ay.ac All Rights Reserved.
    </S.CopyRightParagraph>
    <S.AboutGAParagraph>
      This website uses Google Analytics, a web analytics service by Google.
      Google Analytics uses cookies, which are placed on your computer to help
      the website analyze how you use the site. You can prevent the the
      collection of your data by disabling cookies in your browser.
    </S.AboutGAParagraph>
  </S.Footer>
);
