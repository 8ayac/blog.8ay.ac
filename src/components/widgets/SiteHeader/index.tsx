import { BlogTitle } from '@/src/components/elements/BlogTitle';
import { SocialLinkWithIcon } from '@/src/components/elements/SocialLinkWithIcon';
import { BLOG_SUBTITLE, BLOG_TITLE } from '@/src/constants/site';
import { mq } from '@/src/shared/styles/mediaQuery';
import styled, { StyledComponent } from '@emotion/styled';
import { faTwitter } from '@fortawesome/free-brands-svg-icons';
import React from 'react';

const S: { [key: string]: StyledComponent<any> } = {
  Header: styled.header`
    position: sticky;
    top: 0;
    z-index: 100;

    display: flex;
    align-items: center;

    min-height: 5rem;
    padding: 0.5rem 1.5rem;

    background-color: ${(props) => props.theme.color.siteHeader.header.bg};
    filter: drop-shadow(
      0 0 2px ${(props) => props.theme.color.siteHeader.header.shadow}
    );
    opacity: 0.9;

    ${mq('sm')} {
      min-height: 3.5rem;
    }
  `,

  HeaderInner: styled.nav`
    display: flex;
    justify-content: space-between;
    width: 100%;
    color: white;
  `,

  InnerWrapperFlexContainerDiv: styled.div`
    display: flex;
    align-items: center;
    filter: drop-shadow(
      2px 2px 1px
        ${(props) =>
          props.theme.color.siteHeader.innerWrapperFlexContainerDiv.shadow}
    );
    opacity: 1;
  `,
};
S.TitleWrapper = styled(S.InnerWrapperFlexContainerDiv)``;
S.NavWrapper = styled(S.InnerWrapperFlexContainerDiv)``;

export const SiteHeader: React.FC = () => (
  <S.Header>
    <S.HeaderInner>
      <S.TitleWrapper>
        <BlogTitle title={BLOG_TITLE} subtitle={BLOG_SUBTITLE} />
      </S.TitleWrapper>

      <S.NavWrapper>
        <SocialLinkWithIcon
          url="http://twitter.com/8ayac"
          description="8ayac"
          icon={faTwitter}
        />
      </S.NavWrapper>
    </S.HeaderInner>
  </S.Header>
);
