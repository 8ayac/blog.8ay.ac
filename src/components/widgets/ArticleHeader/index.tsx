import { BodyHeader } from '@/src/components/elements/BodyHeader';
import { DateTime } from '@/src/components/elements/DateTime';
import { CategoryTagList } from '@/src/components/widgets/CategoryTagList';
import { theme } from '@/src/constants/theme';
import { FlexContainerDiv } from '@/src/shared/styles/abstractStyledComponents';
import { getArticlePagePath } from '@/src/shared/utils';
import { Article } from '@/src/types';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { faTags } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const S = {
  DateTimeWrapperDiv: styled(FlexContainerDiv)`
    flex-flow: row wrap;
    align-items: center;
    margin-bottom: 1rem;

    > * {
      margin-right: 1rem;
    }
  `,

  CategoryListWrapperDiv: styled.div`
    margin-top: 2rem;
    margin-bottom: 0.5rem;
  `,
};

export const ArticleHeader: React.FC<{ article: Article }> = ({ article }) => {
  const date = (
    <S.DateTimeWrapperDiv>
      <DateTime date={article.publishedAt} description="Published At" />
      <DateTime date={article.updatedAt} description="Updated At" />
    </S.DateTimeWrapperDiv>
  );
  const tags = (
    <S.CategoryListWrapperDiv>
      <FontAwesomeIcon
        icon={faTags}
        css={css`
          margin-right: 1rem;
          margin-left: 0.5rem;
          font-size: 1em;
          color: ${theme.color.text.primaryLight};
        `}
      />
      <CategoryTagList tags={article.tags as string[]} />
    </S.CategoryListWrapperDiv>
  );

  return (
    <BodyHeader
      title={article.title}
      linkTo={getArticlePagePath(article)}
      beforeTitle={date}
      afterTitle={tags}
    />
  );
};
