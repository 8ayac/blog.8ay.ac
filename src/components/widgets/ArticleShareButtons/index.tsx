import { config } from '@/site.config';
import { Facebook } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Facebook';
import { Hatena } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Hatena';
import { Line } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Line';
import { Pocket } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Pocket';
import { Reddit } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Reddit';
import { Twitter } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Twitter';
import { getArticlePagePath } from '@/src/shared/utils';
import { Article } from '@/src/types';
import styled from '@emotion/styled';
import React from 'react';
import urljoin from 'url-join';

const S = {
  ComponentWrapperSection: styled.section``,

  ButtonListWrapperFlexDiv: styled.div`
    display: flex;
  `,

  ButtonWrapperDiv: styled.div`
    margin-right: 1rem;
    margin-bottom: 0.5rem;
    filter: drop-shadow(
      2px 2px 1px
        ${(props) =>
          props.theme.color.articleShareButtons.buttonWrapperDiv.shadow}
    );

    :hover {
      filter: brightness(120%);
    }
  `,
};

export const ArticleShareButtons: React.FC<{
  article: Article;
  size?: number;
}> = ({ article, size = 32 }) => {
  const shareUrl = urljoin(config.site.rootUrl, getArticlePagePath(article));
  const shareTitle = article.attributes.title;
  const shareButtons = (buttonSize: number) => [
    <Facebook key="facebook" shareUrl={shareUrl} size={buttonSize} />,
    <Twitter
      key="twitter"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Hatena
      key="hatena"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Line
      key="line"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Reddit
      key="reddit"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Pocket
      key="pocket"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
  ];

  return (
    <>
      <S.ComponentWrapperSection>
        <S.ButtonListWrapperFlexDiv>
          {shareButtons(size).map((shareButton) => (
            <S.ButtonWrapperDiv key={shareButton.key}>
              {shareButton}
            </S.ButtonWrapperDiv>
          ))}
        </S.ButtonListWrapperFlexDiv>
      </S.ComponentWrapperSection>
    </>
  );
};
