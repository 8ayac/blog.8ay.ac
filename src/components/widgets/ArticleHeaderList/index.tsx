import { ArticleHeader } from '@/src/components/widgets/ArticleHeader';
import { Article } from '@/src/types';

import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    display: grid;
    grid-gap: 5%;
    margin: 0 auto;
  `,

  ArticleHeaderWrapperDiv: styled.div`
    h1 {
      font-size: 2.3rem;
    }
  `,
};

export const ArticleHeaderList: React.FC<{ articles: Article[] }> = ({
  articles,
}) => (
  <>
    <S.ComponentWrapperDiv>
      <S.ArticleHeaderWrapperDiv>
        {articles
          .sort(
            (a, b) =>
              new Date(a.publishedAt).getTime() -
              new Date(b.publishedAt).getTime()
          )
          .reverse()
          .map((article) => (
            <ArticleHeader key={article.attributes.id} article={article} />
          ))}
      </S.ArticleHeaderWrapperDiv>
    </S.ComponentWrapperDiv>
  </>
);
