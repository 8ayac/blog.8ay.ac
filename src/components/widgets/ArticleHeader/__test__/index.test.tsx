import { ArticleHeader } from '@/src/components/widgets/ArticleHeader';
import { theme } from '@/src/constants/theme';
import { Article } from '@/src/types';
import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';
import React from 'react';

describe('ArticleHeader', () => {
  const testArticleData: Article = {
    attributes: {
      id: 'test-article',
      title: 'Test Title',
      publishedAt: new Date('2000-01-01T00:00:00.000Z'),
      tags: ['test-tag1', 'test-tag2', 'test-tag3'],
    },
    body: `Test Body`,
  };

  describe('is rendered correctly', () => {
    test(' to match the snapshot', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <ArticleHeader article={testArticleData} />
        </ThemeProvider>
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('have proper link title', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <ArticleHeader article={testArticleData} />
        </ThemeProvider>
      );

      expect(wrapper.find('a').first().prop('href')).toBe(
        `/articles/2000-01-01/test-article`
      );
    });
  });

  describe('has proper style rules', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <ArticleHeader article={testArticleData} />
      </ThemeProvider>
    );

    test('in DateTimeWrapperDiv', () => {
      expect(wrapper.find('DateTimeWrapperDiv')).toHaveStyleRule(
        'display',
        'flex'
      );
      expect(wrapper.find('DateTimeWrapperDiv')).toHaveStyleRule(
        'flex-flow',
        'row wrap'
      );
      expect(wrapper.find('DateTimeWrapperDiv')).toHaveStyleRule(
        'align-items',
        'center'
      );
      expect(wrapper.find('DateTimeWrapperDiv')).toHaveStyleRule(
        'margin-bottom',
        '1rem'
      );

      expect(wrapper.find('DateTimeWrapperDiv')).toHaveStyleRule(
        'margin-right',
        '1rem',
        {
          target: '*',
        }
      );
    });

    test('in TagListWrapperDiv', () => {
      expect(wrapper.find('TagListWrapperDiv')).toHaveStyleRule(
        'margin-top',
        '2rem'
      );
      expect(wrapper.find('TagListWrapperDiv')).toHaveStyleRule(
        'margin-bottom',
        '0.5rem'
      );
    });
  });
});
