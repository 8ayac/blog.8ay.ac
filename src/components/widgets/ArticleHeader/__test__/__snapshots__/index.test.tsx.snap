// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeader is rendered correctly  to match the snapshot 1`] = `
.emotion-0 {
  margin-bottom: 3rem;
  border-bottom: 0.25rem solid hsla(0, 0%, 75%);
}

.emotion-2 {
  margin-bottom: 0.3rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.emotion-4>* {
  margin-right: 1rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-8 {
  margin-right: 0.5rem;
  font-weight: 600;
}

.emotion-10 {
  font-weight: bolder;
  color: hsla(212, 72%, 35%);
}

.emotion-18 {
  margin: 0 0 1rem;
}

.emotion-20 {
  margin: 0;
  color: hsla(0, 0%, 15%);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 75%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 75%)
      );
}

.emotion-20:visited {
  color: hsla(0, 0%, 15%);
}

.emotion-22 {
  margin-top: 0.3rem;
}

.emotion-24 {
  margin-top: 2rem;
  margin-bottom: 0.5rem;
}

.emotion-26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-26>* {
  margin-right: 0.7rem;
  margin-bottom: 0.7rem;
}

.emotion-28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsla(0, 0%, 40%);
}

.emotion-30 {
  padding: 0.2rem 1rem;
  font-size: 0.8em;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(212, 72%, 45%);
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(212, 72%, 75%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(212, 72%, 75%)
      );
  border-radius: 3px;
}

.emotion-30:hover {
  color: white;
  background-color: hsla(212, 72%, 55%);
}

.emotion-30:visited {
  color: white;
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "articleShareButtons": Object {
          "buttonWrapperDiv": Object {
            "shadow": "hsla(212, 72%, 75%)",
          },
        },
        "blogTitle": Object {
          "logoWrapperAnchor": Object {
            "textOnHover": "hsla(148, 72%, 75%)",
          },
        },
        "bodyHeader": Object {
          "titleAnchor": Object {
            "shadow": "hsla(148, 72%, 75%)",
          },
        },
        "categoryTag": Object {
          "anchor": Object {
            "bg": "hsla(212, 72%, 45%)",
            "bgOnHover": "hsla(212, 72%, 55%)",
            "shadow": "hsla(212, 72%, 75%)",
            "text": "white",
          },
        },
        "common": Object {
          "border": Object {
            "primary": "hsla(0, 0%, 50%)",
            "primaryLight": "hsla(0, 0%, 75%)",
          },
          "text": Object {
            "accent": "hsla(148, 72%, 45%)",
            "primary": "hsla(0, 0%, 15%)",
            "primaryLight": "hsla(0, 0%, 40%)",
            "secondary": "hsla(212, 72%, 35%)",
          },
        },
        "dateTime": Object {
          "isoDateTime": Object {
            "text": "hsla(212, 72%, 35%)",
          },
        },
        "global": Object {
          "body": Object {
            "bg": "hsla(60, 72%, 55%)",
          },
          "table": Object {
            "tr": Object {
              "bg2n": "hsla(148, 72%, 85%)",
            },
          },
        },
        "markdownRenderer": Object {
          "convertedMdBody": Object {
            "blockquote": Object {
              "border": "hsla(212, 72%, 85%)",
            },
          },
        },
        "siteHeader": Object {
          "header": Object {
            "bg": "hsla(148, 72%, 25%)",
            "shadow": "hsla(148, 72%, 15%)",
          },
          "innerWrapperFlexContainerDiv": Object {
            "shadow": "hsla(148, 72%, 15%)",
          },
        },
        "socialLinkWithIcon": Object {
          "socialLinkAnchor": Object {
            "textOnHover": "hsla(60, 72%, 55%)",
          },
        },
      },
    }
  }
>
  <ArticleHeader
    article={
      Object {
        "attributes": Object {
          "id": "test-article",
          "publishedAt": 2000-01-01T00:00:00.000Z,
          "tags": Array [
            "test-tag1",
            "test-tag2",
            "test-tag3",
          ],
          "title": "Test Title",
        },
        "body": "Test Body",
      }
    }
  >
    <BodyHeader
      afterTitle={
        <TagListWrapperDiv>
          <CategoryTagList
            tags={
              Array [
                "test-tag1",
                "test-tag2",
                "test-tag3",
              ]
            }
            withPreIcon={true}
          />
        </TagListWrapperDiv>
      }
      beforeTitle={
        <DateTimeWrapperDiv>
          <DateTime
            date={2000-01-01T00:00:00.000Z}
            description="Published At"
          />
          <DateTime
            date={2000-01-01T00:00:00.000Z}
            description="Updated At"
          />
        </DateTimeWrapperDiv>
      }
      linkTo="/articles/2000-01-01/test-article"
      title="Test Title"
    >
      <WrapperHeader>
        <header
          className="emotion-0 emotion-1"
        >
          <BeforeTitleDiv>
            <div
              className="emotion-2 emotion-3"
            >
              <DateTimeWrapperDiv>
                <div
                  className="emotion-4 emotion-5"
                >
                  <DateTime
                    date={2000-01-01T00:00:00.000Z}
                    description="Published At"
                  >
                    <ComponentWrapperDiv>
                      <div
                        className="emotion-6 emotion-7"
                      >
                        <DescriptionSpan>
                          <span
                            className="emotion-8 emotion-9"
                          >
                            Published At:
                          </span>
                        </DescriptionSpan>
                        <ISODateTime>
                          <time
                            className="emotion-10 emotion-11"
                          >
                            2000-01-01T00:00:00.000Z
                          </time>
                        </ISODateTime>
                      </div>
                    </ComponentWrapperDiv>
                  </DateTime>
                  <DateTime
                    date={2000-01-01T00:00:00.000Z}
                    description="Updated At"
                  >
                    <ComponentWrapperDiv>
                      <div
                        className="emotion-6 emotion-7"
                      >
                        <DescriptionSpan>
                          <span
                            className="emotion-8 emotion-9"
                          >
                            Updated At:
                          </span>
                        </DescriptionSpan>
                        <ISODateTime>
                          <time
                            className="emotion-10 emotion-11"
                          >
                            2000-01-01T00:00:00.000Z
                          </time>
                        </ISODateTime>
                      </div>
                    </ComponentWrapperDiv>
                  </DateTime>
                </div>
              </DateTimeWrapperDiv>
            </div>
          </BeforeTitleDiv>
          <TitleH1>
            <h1
              className="emotion-18 emotion-19"
            >
              <Link
                href="/articles/2000-01-01/test-article"
                passHref={true}
              >
                <TitleAnchor
                  href="/articles/2000-01-01/test-article"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <a
                    className="emotion-20 emotion-21"
                    href="/articles/2000-01-01/test-article"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    Test Title
                  </a>
                </TitleAnchor>
              </Link>
            </h1>
          </TitleH1>
          <AfterTitleDiv>
            <div
              className="emotion-22 emotion-23"
            >
              <TagListWrapperDiv>
                <div
                  className="emotion-24 emotion-25"
                >
                  <CategoryTagList
                    tags={
                      Array [
                        "test-tag1",
                        "test-tag2",
                        "test-tag3",
                      ]
                    }
                    withPreIcon={true}
                  >
                    <S-ComponentWrapperDiv>
                      <div
                        className="emotion-26 emotion-27"
                      >
                        <S-PreIconWrapperDiv>
                          <div
                            className="emotion-28 emotion-29"
                          >
                            <FontAwesomeIcon
                              border={false}
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    640,
                                    512,
                                    Array [],
                                    "f02c",
                                    "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                                  ],
                                  "iconName": "tags",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-tags fa-w-20 "
                                data-icon="tags"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 640 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                        </S-PreIconWrapperDiv>
                        <CategoryTag
                          key="test-tag1"
                          name="test-tag1"
                        >
                          <Link
                            href="/tags/test-tag1"
                            passHref={true}
                          >
                            <Anchor
                              href="/tags/test-tag1"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="emotion-30 emotion-31"
                                href="/tags/test-tag1"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                test-tag1
                              </a>
                            </Anchor>
                          </Link>
                        </CategoryTag>
                        <CategoryTag
                          key="test-tag2"
                          name="test-tag2"
                        >
                          <Link
                            href="/tags/test-tag2"
                            passHref={true}
                          >
                            <Anchor
                              href="/tags/test-tag2"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="emotion-30 emotion-31"
                                href="/tags/test-tag2"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                test-tag2
                              </a>
                            </Anchor>
                          </Link>
                        </CategoryTag>
                        <CategoryTag
                          key="test-tag3"
                          name="test-tag3"
                        >
                          <Link
                            href="/tags/test-tag3"
                            passHref={true}
                          >
                            <Anchor
                              href="/tags/test-tag3"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="emotion-30 emotion-31"
                                href="/tags/test-tag3"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                test-tag3
                              </a>
                            </Anchor>
                          </Link>
                        </CategoryTag>
                      </div>
                    </S-ComponentWrapperDiv>
                  </CategoryTagList>
                </div>
              </TagListWrapperDiv>
            </div>
          </AfterTitleDiv>
        </header>
      </WrapperHeader>
    </BodyHeader>
  </ArticleHeader>
</ThemeProvider>
`;
